vmdb/app/models/mixins/miq_provision_mixin.rb

```diff
@@ -66,7 +66,10 @@ def get_owner
       email = get_option(:owner_email).try(:downcase)
       return if email.blank?
       requester = miq_request.requester
-      requester.email.to_s.downcase == email ? requester : User.where("lower(email) = ?", email).first
+      owner = requester.email.to_s.downcase == email ? requester : User.where("lower(email) = ?", email).first
+      group_id = get_option(:owner_group)
+      owner.current_group = MiqGroup.where(:description => group_id) if group_id
```

Comment:
We also need to do a little validation to make sure the owner is still associated with the stored group value.  We could break this out into a separate method and do the validation there.  The default if the user is not in the specified group would be to leave the current_group setting as is.  One optimization available here would be to not make any changes if the user is only a member of a single group.